### ISSUE TITLE
Testing any contract including the default greeter no longer works, ABI is None

### ISSUE THREAD

**Issue by SmokinCaterpillar** [ethereum/populus#401]
Hi, great library, I have been using it for a while now, but recently it stopped working (maybe I updated something?).

* Populus Version: 2.0.2
* OS: Linux (Ubuntu 16.04)
* Python web3 Version:  3.16.4
* solc Version: 0.4.19+commit.c4cbbb05.Linux.g++

### What was wrong?

I simply ran `populus init` and `popolus compile` followed by `pytest tests` and
this fails immediately with:

```

    def test_custom_greeting(chain):
>       greeter, _ = chain.provider.get_or_deploy_contract('Greeter')

tests/test_greeter.py:9: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../anaconda3/envs/trump/lib/python3.6/site-packages/populus/contracts/provider.py:143: in get_or_deploy_contract
    deploy_kwargs=deploy_kwargs,
../../anaconda3/envs/trump/lib/python3.6/site-packages/populus/contracts/provider.py:121: in deploy_contract
    kwargs=deploy_kwargs,
../../anaconda3/envs/trump/lib/python3.6/site-packages/web3/contract.py:308: in deploy
    deploy_transaction['data'] = cls._encode_constructor_data(args, kwargs)
../../anaconda3/envs/trump/lib/python3.6/site-packages/eth_utils/string.py:85: in inner
    return force_obj_to_text(fn(*args, **kwargs))
../../anaconda3/envs/trump/lib/python3.6/site-packages/web3/contract.py:741: in _encode_constructor_data
    constructor_abi = get_constructor_abi(cls.abi)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

contract_abi = None

    def get_constructor_abi(contract_abi):
        candidates = [
>           abi for abi in contract_abi if abi['type'] == 'constructor'
        ]
E       TypeError: 'NoneType' object is not iterable

../../anaconda3/envs/trump/lib/python3.6/site-packages/web3/utils/abi.py:230: TypeError
=========================================================== 2 failed in 0.49 seconds ==========================================================
```

#### Cute Animal Picture

![Kitty](https://i1.wp.com/catnamescity.com/wp-content/uploads/2012/08/Cute-Names-for-Cats.jpg?w=233)

**Comment by pipermerriam** [comment]
Temporary fix should be to roll back to `2.0.1`.  Need to investigate why https://github.com/ethereum/populus/pull/399 broke things.

**Comment by SmokinCaterpillar** [comment]
Unfortunately, this does not work either, running `populus compile` on the greeter with `2.0.1` fails with:
```

Traceback (most recent call last):
  File "/home/robert/anaconda3/envs/trump/bin/populus", line 11, in <module>
    sys.exit(main())
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/populus/cli/compile_cmd.py", line 29, in compile_cmd
    compile_project(project, watch)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/populus/api/compile_contracts.py", line 17, in compile_project
    _, compiled_contracts = compile_project_contracts(project)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/populus/compilation/__init__.py", line 54, in compile_project_contracts
    compiled_contracts = post_process_compiled_contracts(base_compiled_contracts)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/populus/utils/compile.py", line 135, in post_process_compiled_contracts
    add_full_dependencies_to_compiled_contracts,
  File "cytoolz/functoolz.pyx", line 586, in cytoolz.functoolz.pipe (cytoolz/functoolz.c:10663)
  File "cytoolz/functoolz.pyx", line 562, in cytoolz.functoolz.c_pipe (cytoolz/functoolz.c:10494)
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/eth_utils/functional.py", line 22, in inner
    return callback(fn(*args, **kwargs))
  File "/home/robert/anaconda3/envs/trump/lib/python3.6/site-packages/populus/utils/compile.py", line 102, in add_direct_dependencies_to_compiled_contracts
    in itertools.chain(contract_data['linkrefs'], contract_data['linkrefs_runtime'])
KeyError: 'linkrefs'
```

**Comment by pipermerriam** [comment]
https://github.com/ethereum/populus/pull/402 should fix. waiting for ci to pass.

**Comment by pipermerriam** [comment]
This *should* be fixed now in `v2.1.0`

**Comment by avatar-lavventura** [comment]
How can I install `Populus v2.1.0? I tried this: sudo pip install populus==2.1.0 but facing with following error: `Command "python setup.py egg_info" failed with error code 1 in /private/tmp/pip-build-F6Rai6/eth-keys/`
